services/py-api/src/database/repository/teams_repository.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from bson import ObjectId
+from motor.motor_asyncio import AsyncIOMotorClientSession
+from pydantic import BaseModel
+from pymongo.errors import DuplicateKeyError
+from result import Result, Err, Ok
+from structlog.stdlib import get_logger
+
+from src.database.db_manager import DatabaseManager
+from src.database.model.team_model import Team
+from src.database.repository.base_repository import CRUDRepository
+from src.server.exception import DuplicateTeamName
+from src.server.schemas.request_schemas.schemas import ParticipantRequestBody
+
+LOG = get_logger()
+
+
+class TeamsRepository(CRUDRepository):
+    def __init__(self, db_manager: DatabaseManager, collection: str) -> None:
+        self._client = db_manager.client
+        self._collection = self._client[db_manager.DB_NAME][collection]
+
+    async def create(
+        self, input_data: ParticipantRequestBody, session: Optional[AsyncIOMotorClientSession] = None
+    ) -> Ok[Team] | Err[DuplicateKeyError] | Err[Exception]:
+
+        try:
+            team = Team(_id=ObjectId(), name=input_data.team_name)
+            LOG.debug("Inserting team {}".format(team.model_dump()))
+            await self._collection.insert_one(document=team.__dict__, session=session)
+            return Ok(team)
+
+        except DuplicateKeyError:
+            LOG.warning("Team insertion failed due to duplicate team name {}".format(input_data.team_name))
+            return Err(DuplicateTeamName("Duplicate team name"))
```

Comment:
`DuplicateTeamNameError` and pass just the team name

Reply:
Fixed in 1a00dd8c6f9016c7770fd6e17ffc6b8195dc8ce2